### DISCUSSION TITLE
Neovim in headless mode shows E492: Not an editor command: PlenaryBustedDirectory

### DISCUSSION THREAD

**Discussion by svenischade** [LazyVim/LazyVim#3218] [Q&A] [answered]
I want to develop a simple plugin and test it using --headless on commandline:

`$ nvim --headless --noplugin -u scripts/minimal_init.vim -c "PlenaryBustedDirectory lua/tests/automated/ { minimal_init = './scripts/minimal_init.vim' }"`

./scripts/minimal_init.vim:
```

set rtp+=.
set rtp+=../plenary.nvim/

runtime! plugin/plenary.vim
runtime! plugin/myplugin.lua

let g:myplugin_test_delay = 100

```
  
This has worked with a neovim installation using _packer_, but after I switched to Lazyvim the following happens:

```
$ nvim --headless --noplugin -u scripts/minimal_init.vim -c "PlenaryBustedDirectory lua/tests/automated/ { minimal_init = './scripts/minimal_init.vim' }"
Error detected while processing command line:
E492: Not an editor command: PlenaryBustedDirectory lua/tests/automated/ { minimal_init = './scripts/minimal_init.vim' }

```  
  
My first guess was that the plenary plugin is loaded _lazy_ so I added the following to my lazyvim config:

~/.config/nvim/lua/plugins/plenary.lua
```
return {
  {
    "nvim-lua/plenary.nvim",
    lazy = false,
    -- enabled = true,    -- does not fix it
    -- priority	= 1000,   -- does not fix it
  },
}
```

Unfortunately, it doesn't fix the problem.

In the Neovim UI the command` :PlenaryBustedDirectory` is recognized. 

What am I doing wrong? Any help is greatly appreciated.

**Comment by folke** [comment, marked as answer]
Check some of my plugins how I run tests.
Like here https://github.com/folke/lazy.nvim/blob/main/tests/init.lua

**Reply by svenischade** [reply]
Thanks for the fast reply. I added your init.lua to my plugin, but now this happens:

```
$ nvim --headless -u lua/tests/init.lua -c "PlenaryBustedDirectory lua/tests {minimal_init = 'lua//tests//init.lua', sequential = true}"

Error detected while processing .../lua/tests/init.lua:
E5113: Error while calling lua chunk: .../lua/tests/init.lua:12: attempt to index field 'uv' (a nil value)
stack traceback:
	.../lua/tests/init.lua:12: in function 'load'
	.../lua/tests/init.lua:29: in function 'setup'
	.../lua/tests/init.lua:36: in main chunk
Error detected while processing command line:
E492: Not an editor command: PlenaryBustedDirectory lua/tests {minimal_init = 'lua//tests//init.lua', sequential = true}

```
The issue is caused within the _load()_ function

```
function M.load(plugin)
...
  local package_root = M.root(".tests/site/pack/deps/start/")
  if not vim.uv.fs_stat(package_root .. name) then             // < === line 12
    print("Installing " .. plugin)
...
end
```
I am using Neovim version:
```
$ nvim -v
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1692716794

```
Seems like s.th. is not initialized or missing...

**Reply by dpetka2001** [reply]
`vim.uv` is only available on Neovim 0.10+. Change that line to `vim.loop` instead.

**Reply by svenischade** [reply]
@dpetka2001 Thank you very much. It works now.
